src/BinaryGapMaras.java

```diff
@@ -0,0 +1,31 @@
+import com.sun.deploy.util.ArrayUtil;
+import com.sun.deploy.util.StringUtils;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+
+public class BinaryGapMaras {
+
+    public static int solution(int n) {
+        String binaryString = Integer.toBinaryString(n);
+        int length;
+        int result = 0;
+
+        String[] ss = binaryString.split("");
+        while (ss[ss.length - 1].equals("0")) {
+            length = ss.length;
+            ss = Arrays.copyOf(ss, length - 1);
```

Comment:
Rozumiem ze probowales skurczyc tablice zeby nie obslugiwac ostatneigo elementu ale ta petla while bedzie miala comp O(n^2) tylko po to zeby nie sprawdzac pozniej w petli warunku czy ostatni element tablicy nie jest 1.